### PR TITLE
feat(react-grid): allow to focus each column and change sorting

### PR DESCRIPTION
**Pull request by MaximKudriavtsev**

### CODE REVIEW

File: packages/dx-react-grid-bootstrap3/src/templates/table-header-cell.test.jsx [hunk 1]
```diff
@@ -232,4 +232,83 @@ describe('TableHeaderCell', () => {
...
+        />
+      ));
+
+      const targetElement = tree.find('div');
+      targetElement.simulate('keydown', { keyCode: ENTER_KEY_CODE, ctrlKey: true });
+      expect(changeSortingDirection)
+        .toHaveBeenCalledWith({ keepOther: true, cancel: true });
+    });
+
+    it('should switch sorting derection when "Enter" key down', () => {
```
Comment (gsobolev) [inline packages/dx-react-grid-bootstrap3/src/templates/table-header-cell.test.jsx]: \* direction

File: packages/dx-react-grid-bootstrap3/src/templates/table-header-cell.jsx [hunk 2]
```diff
@@ -18,13 +21,32 @@ export class TableHeaderCell extends React.PureComponent {
...
       const { allowSorting, changeSortingDirection } = this.props;
       if (!allowSorting) return;
-      e.stopPropagation();
       const cancelSortingRelatedKey = e.metaKey || e.ctrlKey;
       changeSortingDirection({
         keepOther: e.shiftKey || cancelSortingRelatedKey,
         cancel: cancelSortingRelatedKey,
       });
     };
+    this.handleKeyDown = (e) => {
```
Comment (gsobolev) [inline packages/dx-react-grid-bootstrap3/src/templates/table-header-cell.jsx]: This method looks very similar to `onCellClick`. Let's try to refactor the methods mentioned somehow.

File: packages/dx-react-grid-bootstrap3/src/templates/table-header-cell.jsx [hunk 3]
```diff
@@ -18,13 +21,32 @@ export class TableHeaderCell extends React.PureComponent {
     this.onCellClick = (e) => {
       const { allowSorting, changeSortingDirection } = this.props;
       if (!allowSorting) return;
-      e.stopPropagation();
```
Comment (gsobolev) [inline packages/dx-react-grid-bootstrap3/src/templates/table-header-cell.jsx]: Are you sure there is no more need for this?

File: packages/dx-react-grid-bootstrap3/src/templates/table-header-cell.jsx [hunk 4]
```diff
@@ -66,13 +84,18 @@ export class TableHeaderCell extends React.PureComponent {
...
+          onMouseDown={this.handleMouseDown}
+          onBlur={this.handleBlur}
+          onKeyDown={this.handleClick}
         >
           {allowSorting ? (
             <SortingControl
               align={align}
               sortingDirection={sortingDirection}
               columnTitle={columnTitle}
+              handleClick={this.handleClick}
```
Comment (gsobolev) [inline packages/dx-react-grid-bootstrap3/src/templates/table-header-cell.jsx]: Cannot find the definition and usage of this property in the `SortingControl` component.

File: packages/dx-react-grid-material-ui/src/templates/table-header-cell/sorting-control.jsx [hunk 5]
```diff
@@ -1,53 +1,55 @@
...
+    </TableSortLabel>
+  </Tooltip>
+);
 
 SortingControlBase.propTypes = {
   align: PropTypes.string.isRequired,
   sortingDirection: PropTypes.oneOf(['asc', 'desc']),
   columnTitle: PropTypes.string.isRequired,
   classes: PropTypes.object.isRequired,
+  handleClick: PropTypes.func.isRequired,
```
Comment (gsobolev) [inline packages/dx-react-grid-material-ui/src/templates/table-header-cell/sorting-control.jsx]: This name looks weird. Why not use 'onClick'?

File: packages/dx-react-grid-material-ui/src/templates/table-header-cell.test.jsx [hunk 6]
```diff
@@ -1,9 +1,10 @@
 import React from 'react';
-import { TableCell } from 'material-ui';
+import { TableCell, TableSortLabel } from 'material-ui';
 import { createMount, getClasses } from 'material-ui/test-utils';
```
Comment (kvet) [inline packages/dx-react-grid-material-ui/src/templates/table-header-cell.test.jsx]: Maybe we should use shallow rendering here?

**Pull request merged**